internal/codeintel/autoindexing/internal/background/dependencies/utils.go

```diff
@@ -23,16 +25,37 @@ const stalledIndexMaxAge = time.Second * 25
 // "queued" on its next reset.
 const indexMaxNumResets = 3
 
-var IndexWorkerStoreOptions = dbworkerstore.Options[uploadsshared.Index]{
-	Name:              "codeintel_index",
-	TableName:         "lsif_indexes",
-	ViewName:          "lsif_indexes_with_repository_name u",
-	ColumnExpressions: indexColumnsWithNullRank,
-	Scan:              dbworkerstore.BuildWorkerScan(scanIndex),
-	OrderByExpression: sqlf.Sprintf("(u.enqueuer_user_id > 0) DESC, u.queued_at, u.id"),
-	StalledMaxAge:     stalledIndexMaxAge,
-	MaxNumResets:      indexMaxNumResets,
-}
+var (
+	AutoIndexingUseFifoAlgorithm = env.MustGetBool("CODEINTEL_AUTOINDEXING_DEQUEUE_FIFO_ALGORITHM", false, "Use the original FIFO dequeueing algorithm instead of the newer moving-window algorithm.")
+	indexLookbackWindow          = env.Get("CODEINTEL_AUTOINDEXING_DEQUEUE_CANDIDATE_JOB_WINDOW", "1d", "The window from the latest enqueued job to consider when finding candidate jobs to run.")
+	repoDequeueCooldown          = env.Get("CODEINTEL_AUTOINDEXING_DEQUEUE_COOLDOWN_DEBUFF", "6h", "How soon a repository should be on cooldown after a successful index before it can be a candidate when the most recently queued job is outside the lookback window.")
```

Comment:
@efritz do you think theres a way to explain why/how one would tune these according to different situations/requirements? Im half checked-out of this PR already lolmao

Reply:
Something like this?

```suggestion
	repoDequeueCooldown          = env.Get("CODEINTEL_AUTOINDEXING_DEQUEUE_COOLDOWN_DEBUFF", "6h", "The minimum time since the last dequeue for a repository before records outside of the lookback window are made visible for processing. This allows older records of a repo to be processed, but with lower priority than records within the lookback window for other repositories.")
```